mkdocs_simple_plugin/plugin.py

```diff
@@ -105,24 +227,55 @@ def copy_doc_files(self, dest_dir):
         paths = []
         for root, dirs, files in os.walk("."):
             if self.in_include_dir(root):
+                doc_root = dest_dir + root[1:]
                 for f in files:
                     if self.in_extensions(f):
-                        doc_root = dest_dir + root[1:]
-                        orig = "{}/{}".format(root, f)
-                        new = "{}/{}".format(doc_root, f)
-                        try:
-                            os.makedirs(doc_root, exist_ok=True)
-                            shutil.copy(orig, new)
-                            utils.log.debug(
-                                "mkdocs-simple-plugin: {} --> {}".format(orig, new))
-                            paths.append((orig, new))
-                        except Exception as e:
-                            utils.log.warn(
-                                "mkdocs-simple-plugin: error! {}.. skipping {}".format(e, orig))
-
+                        paths.extend(self.copy_file(root, f, doc_root))
+                    else:
+                        paths.extend(self.extract_from(root, f, doc_root))
             dirs[:] = [d for d in dirs if self.in_search_dir(d, root)]
         return paths
 
+    def copy_file(self, from_dir, f, dest_dir):
+        """Copy the file in from_dir named f to the dest_dir"""
+        orig = "{}/{}".format(from_dir, f)
+        if self.in_extensions(f):
+            newfile = "{}/{}".format(dest_dir, f)
+            try:
+                os.makedirs(dest_dir, exist_ok=True)
+                shutil.copy(orig, newfile)
+                utils.log.debug("mkdocs-simple-plugin: {} --> {}".format(
+                    orig, newfile))
+                return [(orig, newfile)]
+            except Exception as e:
+                utils.log.warn("mkdocs-simple-plugin: error! {}.. skipping {}".format(e, orig))
+        return []
+
+    def extract_from(self, from_dir, f, dest_dir):
+        """Extract content from the file in from_dir named f to a file
+           or files in dest_dir, as specified by the semiliterate parameters
+        """
+        newpaths = []
```

Comment:
style: prefer underscores between words

```suggestion
        new_paths = []
```